blocksync/reactor.go

```diff
@@ -396,14 +407,24 @@ FOR_LOOP:
 
 			// See if there are any blocks to sync.
 			first, second, extCommit := bcR.pool.PeekTwoBlocks()
+			if first == nil || second == nil {
+				// we need to have fetched two consecutive blocks in order to
+				// perform blocksync verification
+				continue FOR_LOOP
+			}
+			// Some sanity checks on heights
+			if state.LastBlockHeight > 0 && state.LastBlockHeight+1 != first.Height {
```

Comment:
I'm afraid we can't.
The value of 0 for `state.LastBlockHeight` is a special value: means we don't have any height yet (think of it as sort of a "nil" value)
So what I'm doing here is first guarding against "nil", before using the value.

To explain it with an example, if the first height of a chain is 100 (instead of 1, which is the usual value) this will panic while it shouldn't

Reply:
Understood. Thanks for the explanation.